### DISCUSSION TITLE
Failed to extract shortcode

### DISCUSSION THREAD

**Discussion by okohll** [google/docsy#986] [Q&A] [answered]
Hi,

I'm not sure what I've done, but when I run `hugo` today I'm getting the following error:

```
Start building sites … 
hugo v0.97.2+extended darwin/amd64 BuildDate=unknown
Error: Error building site: "/Users/oliverkohll/Documents/workspace/abdocs/content/_index.html:7:1": failed to extract shortcode: template for shortcode "blocks/cover" not found
Total in 821 ms
```

Line 7 of the file it references is

`{{< blocks/cover title="agileBase documentation" image_anchor="top" height="full" color="orange" >}}`

which hasn't changed recently.

I've tried running `hugo mod get -u github.com/google/docsy` to make sure I'm on the latest Docsy version.

The source is here: https://github.com/okohll/abdocs

Any ideas on how to debug further?

Thanks
Oliver

**Comment by okohll** [comment]
Any help appreciated - hugo has been upgraded to 0.97.3, I've cloned my repository into a new folder and re-checked 

https://www.docsy.dev/docs/get-started/docsy-as-module/start-from-scratch/

but still get

`failed to extract shortcode: template for shortcode "blocks/cover" not found`

Without this I can't deploy the site. I guess I could go back to using Docsy as a git submodule, will try that unless there are any better ideas.

**Reply by deining** [reply]
I cannot repoduce your problem: if I clone your repo and run `hugo server`, everything is fine for me.
Did you try to clear your module via `hugo mod clean --all`?

**Reply by okohll** [reply]
That did it, thanks you!

**Reply by okohll** [reply]
Had to do this again just now, so it may be a regular thing, perhaps on a version upgrade of Hugo or module. Have added it to my script

**Reply by afandian** [reply]
This is happening to me, from a clean image in CI. Using `klakegg/hugo:ext-alpine` per docsy-example's [Dockerfile](https://github.com/google/docsy-example/blob/master/Dockerfile). 

In this case I'm not even using Docsy as a module (getting it via the git submodule since Hugo modules are so flaky on my machine). 
```
$ hugo mod clean --all
Deleted 0 files from module cache.
...
Start building sites … 
hugo v0.105.0-0e3b42b4a9bdeb4d866210819fc6ddcf51582ffa+extended linux/amd64 BuildDate=2022-10-28T12:29:05Z VendorInfo=gohugoio
Error: Error building site: "/builds/myrepo/content/_index.html:6:1": failed to extract shortcode: template for shortcode "blocks/cover" not found
Total in 12 ms
```

This is also showing up in the docsy-example repo: https://github.com/google/docsy-example/issues/41

Any ideas on how to get this to build?

**Reply by afandian** [reply]
I think the solution may be a deeper `git submodule update --init --recursive`.

**Comment by okohll** [comment, marked as answer]
@deining 's suggestion of `hugo mod clean --all` fixed it

**Comment by OzzDizz** [comment]
Am having this issue as well from scratch build, have tried hugo mod clean -all to no avail.

Error building site: "/home/myname/Documents/GitHub/my-new-site/content/en/_index.md:5:1": failed to extract shortcode: template for shortcode "blocks/cover" not found

**Reply by deining** [reply]
What version of hugo you are using? To check your version simply type:

```
hugo version
```

The error posted by you can occur if you are running an outdated version of Hugo. As of docsy theme version v0.7.0, hugo version 0.110.0 or higher is required.

**Reply by juanchax** [reply]
facing the same, run all commands suggested here and in [https://github.com/google/docsy-example/issues/41]( https://github.com/google/docsy-example/issues/41) still getting this error on build 

`failed to extract shortcode: template for shortcode "blocks/cover" not found`

my Hugo version

`% hugo version
hugo v0.139.4+extended+withdeploy darwin/amd64 BuildDate=2024-12-09T17:45:23Z VendorInfo=brew`
